File: Matérias/AEDS 2/BeeCrowd/Godofor.c
#include <stdio.h>
#include <string.h>

int main() {
    int N;
    scanf("%d", &N);

    char vencedor[105] = "";
    int maxPoder = -1, maxMatou = -1, minMorreu = 101;

    for (int i = 0; i < N; i++) {
        char nome[105];
        int poder, matou, morreu;
        scanf("%s %d %d %d", nome, &poder, &matou, &morreu);

        int melhor = 0;
        if (poder > maxPoder) {
            melhor = 1;
        } else if (poder == maxPoder) {
            if (matou > maxMatou) {
                melhor = 1;
            } else if (matou == maxMatou) {
                if (morreu < minMorreu) {
                    melhor = 1;
                } else if (morreu == minMorreu) {
                    if (strcmp(nome, vencedor) < 0) {
                        melhor = 1;
                    }
                }
            }
        }
        if (melhor) {
            strcpy(vencedor, nome);
            maxPoder = poder;
            maxMatou = matou;
            minMorreu = morreu;
        }
    }
    printf("%s\n", vencedor);
    return 0;

File: Matérias/AEDS 2/BeeCrowd/Godofor.java
import java.util.*;

public class Godofor {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int N = sc.nextInt();
        String vencedor = "";
        int maxPoder = -1, maxMatou = -1, minMorreu = 101;
        for (int i = 0; i < N; i++) {
            String nome = sc.next();
            int poder = sc.nextInt();
            int matou = sc.nextInt();
            int morreu = sc.nextInt();

            boolean melhor = false;
            if (poder > maxPoder) {
                melhor = true;
            } else if (poder == maxPoder) {
                if (matou > maxMatou) {
                    melhor = true;
                } else if (matou == maxMatou) {
                    if (morreu < minMorreu) {
                        melhor = true;
                    } else if (morreu == minMorreu) {
                        if (nome.compareTo(vencedor) < 0) {
                            melhor = true;
                        }
                    }
                }
            }
            if (melhor) {
                vencedor = nome;
                maxPoder = poder;
                maxMatou = matou;
                minMorreu = morreu;
            }
        }
        System.out.println(vencedor);
    }
}
